x-pack/libbeat/management/managerV2.go

```diff
@@ -443,24 +460,40 @@ func (cm *BeatV2Manager) reload(units map[unitKey]*client.Unit) {
 	var outputUnit *client.Unit
 	var inputUnits []*client.Unit
 	var stoppingUnits []*client.Unit
+	var errs multierror.Errors
+
 	for _, unit := range units {
-		state, ll, _ := unit.Expected()
-		if ll > lowestLevel {
+		expected := unit.Expected()
+		if expected.LogLevel > lowestLevel {
 			// log level is still used from an expected stopped unit until
 			// the unit is completely removed (aka. fully stopped)
-			lowestLevel = ll
+			lowestLevel = expected.LogLevel
 		}
-		if state == client.UnitStateStopped {
+		if expected.Features != nil {
```

Comment:
Do we disable the feature if `expected.Features == nil` and the current Features configuration is not nil? Is this a case you need to handle, or does the agent always provided a non-nil Features in expected?

Reply:
In the beginning, before any feature flags are set, Agent will send a `nil` value for `expected.Features`.  So this check here guards against that condition, preventing any unnecessary processing.  Thereafter, whenever a feature flag gets set, Agent will only send non-nil values for `expected.Features`; see https://github.com/elastic/elastic-agent/pull/2218/files#diff-5570d8d3d94d053efb7e59bcad53e531bcf3a0ff5b8d3da9cc3334d859f095c9R176-R181

Since `expected.Features` is built to contain multiple features' configuration in it, we would only disable a feature if that particular feature's configuration within `expected.Features` indicated as such.  For example, if `expected.Features.FQDN.Enabled` was set to `false`.  AFAIU we have not designed for a use case of mass-disabling all features by setting `expecting.Features` to `nil`.

Reply:
> AFAIU we have not designed for a use case of mass-disabling all features by setting expecting.Features to nil.

That doesn't mean someone won't make that edit in the policy. IMO the most obvious thing to do is to translate that to disabling all features, that maps to what happens when you remove an input from the policy. We can file a follow up issue to deal with this, it is definitely an edge case.